release notes/v0.41.0.md

```diff
@@ -0,0 +1,64 @@
+- [#2700](https://github.com/grafana/k6/issues/2700) added support for loading gRPC protoset files. Thanks, @jklipp!
+
+## Bug fixes
+
+- [#2678](https://github.com/grafana/k6/pull/2678) fixed the Docker image labels. Thanks, @knittl, for reporting the problem ([#2677](https://github.com/grafana/k6/issues/2677))!
+- [#2689](https://github.com/grafana/k6/pull/2689) fixed the REST API's `Content-Type` response header. Thanks, @wingyplus!
+- [#2691](https://github.com/grafana/k6/pull/2691) fixed the detailed `k6 version` information embedded in the k6 releases.
+- [#2693](https://github.com/grafana/k6/pull/2693) fixed a bug in the k6 event loop when a Promise rejection was left unhandled.
+- [#2696](https://github.com/grafana/k6/pull/2696) fixed a problem with HTTP redirects with empty `Location` headers ([#2474](https://github.com/grafana/k6/issues/2474)) by updating the Go version we use to compile k6 to 1.19.x. Thanks, @agilob!
+- [#2705](https://github.com/grafana/k6/pull/2705) fixed a panic in the `k6/net/grpc` module ([#2661](https://github.com/grafana/k6/issues/2661)). Thanks, @c47gao and @robpickerill!
+- [#2738](https://github.com/grafana/k6/pull/2738) fixed a panic when a Promise was rejected with an `undefined` reason.
+- [#2739](https://github.com/grafana/k6/pull/2739) fixed hidden stack traces in certain types of errors originating from k6's Go code.
+
+## Maintenance and internal improvements
+
+We had a few minor changes in this release:
+
+- [#2687](https://github.com/grafana/k6/pull/2687) improved our logging tests. Thanks, @nsmith5!
+- [#2696](https://github.com/grafana/k6/pull/2696) updated the used Go version to 1.19.x and the Alpine version in our Docker image to 3.16. Thanks, @agilob!
+- [#2707](https://github.com/grafana/k6/pull/2707), [#2708](https://github.com/grafana/k6/pull/2708), [#2709](https://github.com/grafana/k6/pull/2709), [#2710](https://github.com/grafana/k6/pull/2710) updated most of the Go dependencies k6 has.
+- [#2716](https://github.com/grafana/k6/pull/2716) refactored how custom JS tags are applied to metrics and cleaned up validation for invalid tag values.
+
+As well as a couple of significant improvements that will allow us to develop exciting new features soon:
+
+### Metric time series ([#2594](https://github.com/grafana/k6/pull/2594))
+
+Previous to [#2594](https://github.com/grafana/k6/pull/2594), k6 didn't have an efficient way to group metric samples with the same tags. It meant that a whole class of applications for processing and aggregating metrics were nearly impossible to do or, at best, very inefficient.
+
+At the cost of some minor [breaking changes](#breaking-changes), we now have a performant internal representation to group metric samples with the same tags at the time of the action that generated them, i.e. the time of metric measurement. It allows us to efficiently group samples for the same action (e.g. an HTTP request to a specific URL) over time and construct time series with them.
```

Comment:
I feel like this likely needs more information on what that means for the future possibilities. Even if we don't intend on currently pursuing all of them.

Reply:
Sure, but I think it should be in the "Roadmap and future plans" section, not here

Reply:
Resolved with the [latest force-push](https://github.com/grafana/k6/compare/0b6ec90828afac81746a6d300c878006c57b5ac7..f68483e91bdb7c70579acea6ada26a549c7a6e6e)?